src/pages/execution/tabLoader.tsx

```diff
@@ -255,6 +265,76 @@ export default function tabLoader(tabBase: TabBase): TabData {
   };
 }
 
+function AlertComponent() {
+  const executionAndTasks = useContext(TileFactoryContext).executionAndTasks;
+  const { severity, setSeverity, setHasAlerts } =
+    useContext(TileFactoryContext);
+  const alerts = useMemo(() => {
+    const allAlerts: AlertItem[] = [];
+
+    rules.forEach((rule) => {
+      const [ruleAlerts, maxSeverity] = rule(executionAndTasks, severity);
+      allAlerts.push(...ruleAlerts);
+      setSeverity(maxSeverity);
+    });
+
+    return allAlerts;
+  }, [executionAndTasks, severity, setSeverity]);
+
+  if (alerts.length === 0) {
+    return null;
+  }
+  setHasAlerts(true);
+
+  return (
+    <Stack sx={{ margin: "15px" }} spacing={2}>
+      {alerts.map((alert, index) => (
+        <React.Fragment key={index}>{alert.component}</React.Fragment>
+      ))}
+    </Stack>
+  );
+}
+
+function SummaryTabHead() {
+  const { severity, hasAlerts } = useContext(TileFactoryContext);
+
+  if (!hasAlerts) {
+    return (
+      <div>
+        <span style={{ marginRight: "5px" }}>Summary</span>
+      </div>
+    );
+  }
+
+  let dotColor;
+  switch (severity) {
+    case "ERROR":
+      dotColor = "red";
+      break;
+    case "WARNING":
+      dotColor = "orange";
+      break;
+    case "INFO":
+    default:
+      dotColor = "blue";
+      break;
+  }
+
+  return (
+    <div style={{ display: "flex", alignItems: "center" }}>
+      <span style={{ marginRight: "5px" }}>Summary</span>
+      <div
```

Comment:
For conciseness you could do
`{ !!severity && <div ..../>}` here instead of outputting the `div` in 2 different blocks of code.